tournamentcontrol/competition/tests/test_utils.py

```diff
@@ -124,3 +129,79 @@ def test_sum_dict_integer(self):
 
     def test_revpow(self):
         self.assertEqual(utils.revpow(2**4, 2), 4)
+
+
+class StageGroupPositionTests(TestCase):
+    """
+    Tests for the stage_group_position function.
+    """
+
+    def test_stage_group_position_with_no_pools_raises_index_error(self):
+        """
+        Test that stage_group_position raises IndexError when trying to access
+        a pool that doesn't exist on a stage.
+        """
+        # Create a division with multiple stages so we can reference one
+        division = DivisionFactory.create()
+        stage1 = StageFactory.create(division=division, order=1)
+        stage2 = StageFactory.create(division=division, order=2)
+
+        # Stage1 has no pools, but we try to access group 1 from stage1
+        # This should raise IndexError because stage1.pools.all() is empty
+        # and we try to access index 0 (int("1") - 1)
+        with self.assertRaisesRegex(
+            IndexError,
+            r"Invalid group 1 for stage .* \(stage has 0 pools\)"
+        ):
+            utils.stage_group_position(stage2, "S1G1P1")
+
+    def test_stage_group_position_with_insufficient_pools_raises_index_error(self):
+        """
+        Test that stage_group_position raises IndexError when trying to access
+        a pool index that is out of range.
+        """
+        # Create a division with multiple stages
+        division = DivisionFactory.create()
+        stage1 = StageFactory.create(division=division, order=1)
+        stage2 = StageFactory.create(division=division, order=2)
+
+        # Stage1 has only 1 pool
+        StageGroupFactory.create(stage=stage1)  # Creates pool 1
+
+        # Try to access group 2 from stage1 (which doesn't exist)
+        # This should raise IndexError because stage1.pools.all() has only 1 item
+        # and we try to access index 1 (int("2") - 1)
+        with self.assertRaisesRegex(
+            IndexError,
+            r"Invalid group 2 for stage .* \(stage has 1 pools\)"
+        ):
+            utils.stage_group_position(stage2, "S1G2P1")
+
+    def test_stage_group_position_with_valid_pool_succeeds(self):
+        """
+        Test that stage_group_position works correctly when the pool exists.
```

Comment:
Extend this test.

Add a third stage which "follows" `stage1` such that both `stage2` and `stage3` lookups will refer to `stage1`.

Reply:
Need to make the tests work with and without the S1 in the formula. Please do these as subTest.